src/base-with-proxy.js

```diff
@@ -15,24 +15,41 @@
 define( [
             'jquery',
             'ju-components/base',
-            'ju-components/base-with-proxy/payload-handler'
+            'ju-components/base-with-proxy/payload-handler',
+            'ju-components/resource/storage/template-storage'
         ],
         function (
             $,
             BaseComponent,
-            PayloadHandler
+            PayloadHandler,
+            TemplateStorage
         ) {
 
     'use strict';
 
     var BaseWithProxy = BaseComponent.extend({
-        init : function () {
+        init : function (args, templatePath, resourcesDef, childrenDef) {
```

Comment:
We were talking about the possibility of separating the 'proxy' functionality from this class to a binder that could be applied to the 'base-ui' and the currently 'base-with-save' clases.